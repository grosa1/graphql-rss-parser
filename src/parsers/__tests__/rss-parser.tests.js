/* eslint-env jest */
const parse = require('../rss-parser')
const mockRequest = require('../../__tests__/mockRequests')

test('should parse string from rolflekang.com/feed.xml', async () => {
  const fixture = await mockRequest('https://rolflekang.com/feed.xml')

  expect(await parse(fixture.text)).toMatchSnapshot()
})

test.skip('should parse string from google.blogspot.com/feeds/posts/default', async () => {
  const fixture = await mockRequest('http://google.blogspot.com/feeds/posts/default')

  expect(await parse(fixture.text)).toMatchSnapshot()
})
