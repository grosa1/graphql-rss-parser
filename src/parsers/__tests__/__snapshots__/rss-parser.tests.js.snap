// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse string from google.blogspot.com/feeds/posts/default 1`] = `
Object {
  "entries": Array [
    Object {
      "author": "Brett Wiltshire",
      "content": "<i>Posted by Brett Wiltshire, Blogger CEO</i><br /><br />This morning we’re beyond thrilled to announce that Blogger has signed a definitive agreement to be acquired by Google, the Internet search company. This is exciting news not only for all of us on the Blogger team, but for our users, our partners, and most importantly -- the blogosphere itself. <br /><br />Understandably, you probably have lots of questions about what this means for Blogger and Blogger users. Below, we've put together some initial answers to many of the biggest questions. More info will be available as we figure it out. Thanks for your support as we transfer into this next exciting phase. &nbsp; <br /><br /><br /><b>Q: Why did Blogger sell to Google?</b><br />A: Well, on the surface, it may look obvious: A company of Google's size could give Blogger the resources we needed to do things better, faster, bigger. It's been a long eleven+ years since we started the company, and not all of them were very fun. We had been making serious progress over the last year or so, but bootstrapping (growing without funding) is always a slow process, and we were a long way from where we wanted to be. We wanted to offer a better service. And with Google, it wasn't just their size, of course. They had clearly done an incredible job building their technology and business, and we'd been big fans. <br /><br />However, that doesn't mean it was an easy decision. We'd seen many small companies doing interesting things die (or at least become uninteresting) after getting acquired. It was only after becoming convinced that: a) There were sensible, cool, powerful things we could do on the technology/product side with Google that we couldn't do otherwise; and b) It was a good company, run by people we liked, who wanted the same things we did (i.e., they wouldn't screw up Blogger or make our lives miserable). <br /><br />We became convinced both of these were the case rather quickly after thinking about the product side and talking to lots of Googlers. &nbsp; Also, Google liked our logo. And we liked their food. <br /><br /><b>Q: Will Blogger go away?&nbsp;</b><br />A: Nope. Blogger is going to maintain its branding and services. While we may integrate with Google in certain areas there will always be a Blogger. <br /><br /><b>Q: What does the acquisition mean to Blogger users?&nbsp;</b><br />A: Greater reliability, new innovative products and a whole lot more that we aren't ready to share quite yet (truth is, we're still figuring a lot of it out). Right now, we're mostly bolstering up our hardware -- making things faster and more reliable -- and getting settled. Next, we're going to be working on some killer ideas we've been wanting to implement for years. It'll be fun. <br /><br /><b>Q: Will there be any changes to my account?&nbsp;</b><br />A: Not right now but if anything does change we will notify you ahead of time, as we've done in the past.<br /><br /><b>Q: Will there be any changes to the Blogger products?&nbsp;</b><br />A: We will be making some changes in our product line. We've been working on a new version of Blogger for some time now that will be coming out soon. We'll tell you more as soon as we know. <br /><br /><b>Q: What are your plans for the future?&nbsp;</b><br />A: We are building a nuclear powered... Wait, you almost had us. We aren't telling, yet! But we will have more in a few weeks. <br /><br /><b>Q: Does this mean my blog will rank higher in Google search results?&nbsp;</b><br />A: Nope. It does mean your blog might be stored physically closer to Google but that's about it. The people at Google have done a great job over the years making sure their search results are honest and objective and there's no reason they would change that policy for Blogger or anyone else. <br /><br /><b>Q: What will happen to all the nice kids that work on Blogger?&nbsp;</b><br />A: We’ll still be working on Blogger and making it better. <br /><br /><b>Q: Are you still as handsome as ever?&nbsp;</b><br />A:<br /><div class=\\"separator\\" style=\\"clear: both; text-align: center;\\"><a href=\\"http://1.bp.blogspot.com/_UUTay7LUoq0/TXVACrMZNOI/AAAAAAAAADI/_WWawmSQnIM/s1600/_MG_1538.jpg\\" imageanchor=\\"1\\" style=\\"margin-left: 1em; margin-right: 1em;\\"><img border=\\"0\\" src=\\"http://1.bp.blogspot.com/_UUTay7LUoq0/TXVACrMZNOI/AAAAAAAAADI/_WWawmSQnIM/s400/_MG_1538.jpg\\" width=\\"500\\" /></a></div>",
      "contentSnippet": "Posted by Brett Wiltshire, Blogger CEOThis morning we’re beyond thrilled to announce that Blogger has signed a definitive agreement to be acquired by Google, the Internet search company. This is exciting news not only for all of us on the Blogger team, but for our users, our partners, and most importantly -- the blogosphere itself. Understandably, you probably have lots of questions about what this means for Blogger and Blogger users. Below, we've put together some initial answers to many of the biggest questions. More info will be available as we figure it out. Thanks for your support as we transfer into this next exciting phase. &nbsp; Q: Why did Blogger sell to Google?A: Well, on the surface, it may look obvious: A company of Google's size could give Blogger the resources we needed to do things better, faster, bigger. It's been a long eleven+ years since we started the company, and not all of them were very fun. We had been making serious progress over the last year or so, but bootstrapping (growing without funding) is always a slow process, and we were a long way from where we wanted to be. We wanted to offer a better service. And with Google, it wasn't just their size, of course. They had clearly done an incredible job building their technology and business, and we'd been big fans. However, that doesn't mean it was an easy decision. We'd seen many small companies doing interesting things die (or at least become uninteresting) after getting acquired. It was only after becoming convinced that: a) There were sensible, cool, powerful things we could do on the technology/product side with Google that we couldn't do otherwise; and b) It was a good company, run by people we liked, who wanted the same things we did (i.e., they wouldn't screw up Blogger or make our lives miserable). We became convinced both of these were the case rather quickly after thinking about the product side and talking to lots of Googlers. &nbsp; Also, Google liked our logo. And we liked their food. Q: Will Blogger go away?&nbsp;A: Nope. Blogger is going to maintain its branding and services. While we may integrate with Google in certain areas there will always be a Blogger. Q: What does the acquisition mean to Blogger users?&nbsp;A: Greater reliability, new innovative products and a whole lot more that we aren't ready to share quite yet (truth is, we're still figuring a lot of it out). Right now, we're mostly bolstering up our hardware -- making things faster and more reliable -- and getting settled. Next, we're going to be working on some killer ideas we've been wanting to implement for years. It'll be fun. Q: Will there be any changes to my account?&nbsp;A: Not right now but if anything does change we will notify you ahead of time, as we've done in the past.Q: Will there be any changes to the Blogger products?&nbsp;A: We will be making some changes in our product line. We've been working on a new version of Blogger for some time now that will be coming out soon. We'll tell you more as soon as we know. Q: What are your plans for the future?&nbsp;A: We are building a nuclear powered... Wait, you almost had us. We aren't telling, yet! But we will have more in a few weeks. Q: Does this mean my blog will rank higher in Google search results?&nbsp;A: Nope. It does mean your blog might be stored physically closer to Google but that's about it. The people at Google have done a great job over the years making sure their search results are honest and objective and there's no reason they would change that policy for Blogger or anyone else. Q: What will happen to all the nice kids that work on Blogger?&nbsp;A: We’ll still be working on Blogger and making it better. Q: Are you still as handsome as ever?&nbsp;A:",
      "id": "tag:blogger.com,1999:blog-2120328063286836889.post-7340716506563491347",
      "link": "http://google.blogspot.com/2011/04/google-to-acquire-blogger.html",
      "pubDate": "2011-04-01T15:13:07.987Z",
      "title": "Google to Acquire Blogger",
    },
  ],
  "feedUrl": "http://www.blogger.com/feeds/2120328063286836889/posts/default",
  "link": "http://google.blogspot.com/feeds/posts/default",
  "parser": "RSS_PARSER",
  "title": "Google on BlogSpot",
}
`;

exports[`should parse string from rolflekang.com/feed.xml 1`] = `
Object {
  "description": "",
  "entries": Array [
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/lets-run-some-races",
      "link": "http://rolflekang.com/lets-run-some-races",
      "pubDate": "2017-01-02T00:00:00.000Z",
      "title": "Let's run some races this year",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/software-development-on-an-ipad",
      "link": "http://rolflekang.com/software-development-on-an-ipad",
      "pubDate": "2017-01-01T00:00:00.000Z",
      "title": "Software development on an iPad",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/filtering-lint-errors",
      "link": "http://rolflekang.com/filtering-lint-errors",
      "pubDate": "2016-06-10T00:00:00.000Z",
      "title": "Filtering lint errors",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/interrail-summer",
      "link": "http://rolflekang.com/interrail-summer",
      "pubDate": "2015-07-26T00:00:00.000Z",
      "title": "Interrail summer",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/writing-latex-in-atom",
      "link": "http://rolflekang.com/writing-latex-in-atom",
      "pubDate": "2015-05-23T00:00:00.000Z",
      "title": "Writing (latex) in Atom",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/rmoq",
      "link": "http://rolflekang.com/rmoq",
      "pubDate": "2014-12-30T00:00:00.000Z",
      "title": "rmoq - a request mock cache",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/django-nopassword-one-point-o",
      "link": "http://rolflekang.com/django-nopassword-one-point-o",
      "pubDate": "2014-10-03T15:35:00.000Z",
      "title": "django-nopassword reached 1.0",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/moving-to-pelican",
      "link": "http://rolflekang.com/moving-to-pelican",
      "pubDate": "2014-05-07T00:00:00.000Z",
      "title": "Moving the blog to pelican",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/building-tumblr-theme-with-grunt",
      "link": "http://rolflekang.com/building-tumblr-theme-with-grunt",
      "pubDate": "2014-02-02T00:00:00.000Z",
      "title": "Building a Tumblr theme with GruntJS",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/backup-routines-for-redisdb",
      "link": "http://rolflekang.com/backup-routines-for-redisdb",
      "pubDate": "2013-10-24T00:00:00.000Z",
      "title": "Backup routine for Redis",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/django-app-custom-user",
      "link": "http://rolflekang.com/django-app-custom-user",
      "pubDate": "2013-05-22T00:00:00.000Z",
      "title": "Making your Django app ready for custom users",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/stats-are-fun",
      "link": "http://rolflekang.com/stats-are-fun",
      "pubDate": "2013-03-03T00:00:00.000Z",
      "title": "Stats are fun",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/alfred-scripts-for-jekyll",
      "link": "http://rolflekang.com/alfred-scripts-for-jekyll",
      "pubDate": "2013-02-26T00:00:00.000Z",
      "title": "Alfred scripts for Jekyll",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/use-array-in-postgresql-with-django",
      "link": "http://rolflekang.com/use-array-in-postgresql-with-django",
      "pubDate": "2013-01-27T00:00:00.000Z",
      "title": "Use array in postgresql with django",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/apply-puppet-configuration-automatically",
      "link": "http://rolflekang.com/apply-puppet-configuration-automatically",
      "pubDate": "2013-01-03T00:00:00.000Z",
      "title": "Apply puppet automatically",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/reading-wikipedia",
      "link": "http://rolflekang.com/reading-wikipedia",
      "pubDate": "2012-12-02T00:00:00.000Z",
      "title": "Readable Wikipedia",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/django-nopassword",
      "link": "http://rolflekang.com/django-nopassword",
      "pubDate": "2012-09-28T00:00:00.000Z",
      "title": "django-nopassword",
    },
    Object {
      "content": "",
      "contentSnippet": "",
      "guid": "http://rolflekang.com/the-github-lamp",
      "link": "http://rolflekang.com/the-github-lamp",
      "pubDate": "2012-09-16T00:00:00.000Z",
      "title": "The Github lamp",
    },
  ],
  "feedUrl": "http://rolflekang.com/feed.xml",
  "generator": "rl.com v1.0.0",
  "link": "http://rolflekang.com/",
  "parser": "RSS_PARSER",
  "title": "Writings of Rolf Erik Lekang",
}
`;
